src/remotestorage.js

```diff
@@ -172,20 +172,31 @@ RemoteStorage.prototype = {
   },
 
   /**
-   * TODO: document
+   * Initiate the OAuth authorization flow.
    *
-   * @param {string} authUrl
-   * @param {string} cordovaRedirectUri
+   * @param {object} options
+   * @param {string} [options.authURL] - URL to the authorization endpoint
+   * @param {string} [options.scope] - access scope
+   * @param {string} [options.redirectUri] - client URL that should be redirected
+   *                                         to after authorization
+   * @param {string} [options.clientId] - client identifier (defaults to the
+   *                                      origin of the redirectUri)
    */
-  authorize: function authorize(authURL, cordovaRedirectUri) {
+  authorize: function authorize (options) {
     this.access.setStorageType(this.remote.storageType);
-    var scope = this.access.scopeParameter;
+    if (typeof options.scope === 'undefined') {
+      options.scope = this.access.scopeParameter;
+    }
 
-    var redirectUri = globalContext.cordova ? cordovaRedirectUri : String(Authorize.getLocation());
+    if (typeof options.redirectUri === 'undefined') {
+      options.redirectUri = globalContext.cordova ? config.cordovaRedirectUri : String(Authorize.getLocation());
+    }
```

Comment:
This changes the logic, and I think in a bad way. It's valid to set the redirect URI in an app that is both used in Cordova and outside of it. The original `if` condition was correct in that sense, but this one would not set the Cordova redirect URI, if the normal one is set.

Ironically, among the new tests, there's no test for setting the Cordova redirect URI.

Reply:
I changed it and added the test.